### ISSUE TITLE
Use terraform version syntax like TFEnv

### ISSUE THREAD

**Issue by Nargonath** [hashicorp/setup-terraform#27]
It would be nice to have the terraform action to use the version requirement stated in the `terraform {}` block. Something like TFEnv where we could state `min-version` for the version to use and it would refer to that value for the installation.

Perhaps we can even use TFEnv behind the scene in this action to not reinvent the wheel.

**Comment by Nargonath** [comment]
I just thought that perhaps the solution is just to install and use TFEnv in subsequent step after this action but I wasn't sure it wouldn't mess up with the wrapper installed by this action. Plus I also felt it has its place in this action to be honest.

**Comment by phr3nzii** [comment]
I'd like this functionality too. We currently use the following
```
terraform {
  required_version = "~> 0.12"
}
```
It would be nice to not have to update the GitHub action workflow with each minor version release of 0.12 and using latest would automatically switch us to 0.13 when that is released, not ideal as there may be breaking changes.

**Comment by Nargonath** [comment]
@phr3nzii I believe you could use:

```HCL
terraform {
  required_version = "~> 0.12.0"
}
```
if you want all version `>= 0.12.0` and `< 0.13.0`

**Comment by phr3nzii** [comment]
What I have works fine for restricting versions to any 0.12.x version when using Terraform or tools like tfswitch. However, at the moment I have to update this part of the GitHub Actions workflow with each minor version
```
env:
  tf_version: 0.12.28
```

**Comment by Nargonath** [comment]
Yeah that's what I noticed on my part as well. That'd be great if we could have the information in a single place rather than split in multiple files.

**Comment by MPV** [comment]
Yeah, would be wonderful being able to combine this with tfupdate for running validate & plan on the PRs it creates (for automatically updating terraform and plugins to newer versions):

https://github.com/marketplace/actions/tfupdate-github-actions

**Comment by briceburg** [comment]
duplicate of https://github.com/hashicorp/setup-terraform/issues/407 ?
